homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,232 @@
+"""Cover entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from motionblindsble.const import MotionBlindType, MotionRunningType
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityDescription,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_BLIND_TYPE, CONF_MAC_CODE, DOMAIN, ICON_VERTICAL_BLIND
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class MotionCoverEntityDescription(CoverEntityDescription):
+    """Entity description of a cover entity with default values."""
+
+    key: str = CoverDeviceClass.BLIND.value
+    translation_key: str = CoverDeviceClass.BLIND.value
+
+
+SHADE_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.SHADE
+)
+BLIND_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.BLIND
+)
+CURTAIN_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.CURTAIN
+)
+VERTICAL_ENTITY_DESCRIPTION = MotionCoverEntityDescription(
+    device_class=CoverDeviceClass.CURTAIN, icon=ICON_VERTICAL_BLIND
+)
+
+COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
+    MotionBlindType.HONEYCOMB.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.ROMAN.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.ROLLER.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.DOUBLE_ROLLER.name: SHADE_ENTITY_DESCRIPTION,
+    MotionBlindType.VENETIAN.name: BLIND_ENTITY_DESCRIPTION,
+    MotionBlindType.VENETIAN_TILT_ONLY.name: BLIND_ENTITY_DESCRIPTION,
+    MotionBlindType.CURTAIN.name: CURTAIN_ENTITY_DESCRIPTION,
+    MotionBlindType.VERTICAL.name: VERTICAL_ENTITY_DESCRIPTION,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up blind based on a config entry."""
+
+    blind_class = BLIND_TYPE_TO_CLASS[entry.data[CONF_BLIND_TYPE].upper()]
+    device = hass.data[DOMAIN][entry.entry_id]
+    entity = blind_class(device, entry)
+
+    async_add_entities([entity])
+
+
+class MotionblindsBLECoverEntity(MotionblindsBLEEntity, CoverEntity):
+    """Representation of a blind."""
+
+    _attr_is_closed: bool | None = None
+    _attr_name = None
+
+    def __init__(self, device: MotionDevice, entry: ConfigEntry) -> None:
+        """Initialize the blind."""
+        _LOGGER.debug(
```

Comment:
Please remove side effects from init

Reply:
Moved the callback registers to async_added_to_hass

Reply:
Please remove the logging in `__init__`

Reply:
That would remove the entire init? Would it not be useful to debug log what blind type is created with what class?

Reply:
We don't allow logging side effects in `__init__`.  You could log it where the entity is created instead.

Reply:
What about in the async_added_to_hass?

Reply:
I included the log in `async_added_to_hass`